### PR TITLE
add method to get the mutability of an AllocId

### PR DESCRIPTION
**Pull request by RalfJung**
Miri needs this for https://github.com/rust-lang/miri/issues/2463.

### CODE REVIEW

**Comment by RalfJung** [comment]
@lcnr maybe I could ask you to take a look at this one?

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -628,6 +628,10 @@ impl Step for Miri {
             cargo.env("MIRI_HOST_SYSROOT", sysroot);
             cargo.env("RUSTC_LIB_PATH", builder.rustc_libdir(compiler));
             cargo.env("MIRI", miri);
+            // propagate --bless
+            if builder.config.cmd.bless() {
+                cargo.env("MIRI_BLESS", "Gesundheit");
+            }
```
Comment (jackh726) [inline src/bootstrap/test.rs:634]: This seems unrelated?

Comment (RalfJung) [reply src/bootstrap/test.rs:634]: Yeah it is. It makes Miri's test suite bless work with `x.py --bless`. Didn't seem significant enough for its own PR, but I can split it if you prefer.
